integration/spark/shared/src/main/java/io/openlineage/spark/agent/filters/EventFilterUtils.java

```diff
@@ -28,7 +30,22 @@ public static boolean isDisabled(OpenLineageContext context, SparkListenerEvent
             new SparkNodesFilter(context),
             new CreateViewFilter(context),
             new AdaptivePlanEventFilter(context))
-        .anyMatch(filter -> filter.isDisabled(event.getClass().cast(event)));
+        .anyMatch(
+            filter -> {
+              boolean isDisabled = filter.isDisabled(event.getClass().cast(event));
+              if (isDisabled) {
```

Comment:
Could you please enclose the whole block within `if (log.isDebugEnabled()`? 
This makes sure no method calls are performed when information is not needed.